### PR TITLE
Migrate signing away from rust-sodium

### LINKED ISSUE #1079: Migrate signing away from rust-sodium
Remove the use of `rust_sodium::crypto::sign::*`. It is used as a part of the `AppKeys` and `ClientKeys` struct. It's use is redundant as of #1058. Signing will be done by `AppFullID` and `ClientFullId`. The relevant functions in `safe_app::ffi::crypto` should also be updated to use BLS keys for signing.

### PR DESCRIPTION
**Pull request by lionel1704**
closes #1079

### CODE REVIEW

File: safe_core/src/utils/seed.rs [hunk 1]
```diff
@@ -6,22 +6,11 @@
...
-/// Calculate sign key from seed.
-pub fn sign_pk_from_seed(seed: &str) -> Result<sign::PublicKey, CoreError> {
-    let arr = divide_seed(seed)?;
-    let id_seed = Seed(sha3_256(arr[SEED_SUBPARTS - 2]));
-    let maid_keys = ClientKeys::new(Some(&id_seed));
-    Ok(maid_keys.sign_pk)
-}
-
 /// Divide `seed` into the number of subparts given by `SEED_SUBPARTS`.
 pub fn divide_seed(seed: &str) -> Result<[&[u8]; SEED_SUBPARTS], CoreError> {
```
Comment (m-cat) [inline safe_core/src/utils/seed.rs]: I think it should be possible to deprecate this entire file.

File: safe_app/src/tests/append_only_data.rs [hunk 2]
```diff
@@ -105,9 +105,9 @@ fn managing_permissions_for_an_app() {
             let client5 = client.clone();
             let client6 = client.clone();
 
-            let sign_pk = client.public_key();
+            let app_pk = client.public_key();
```
[the remark this answers is not in the document]
Comment (lionel1704) [reply safe_app/src/tests/append_only_data.rs]: Yes I have. I just did another re-check to verify too. Everything seems right.
We can consider renaming those variables. But that can be in a separate PR.

File: safe_authenticator/src/client.rs [hunk 3]
```diff
@@ -80,7 +79,8 @@ impl AuthClient {
 where {
```
Comment (m-cat) [inline safe_authenticator/src/client.rs]: Can we remove this `where`? I'm surprised the compiler doesn't complain about it 😄

Comment (lionel1704) [reply safe_authenticator/src/client.rs]: I'll sneak this change in a different PR :)

**Pull request merged**